src/ol/layer/Graticule.js

```diff
@@ -484,9 +483,11 @@ class Graticule extends VectorLayer {
     // extents may be passed in different worlds, to avoid endless loop we use only one
     const realExtent = extent.slice();
     if (this.projection_ && this.getSource().getWrapX()) {
-      wrapX(realExtent, this.projection_);
+      wrapExtentX(realExtent, this.projection_);
     }
-    if (this.loadedExtent_ && !containsExtent(bufferExtent(this.loadedExtent_, resolution / 2), realExtent)) {
+    realExtent[0] = Math.round(realExtent[0] * 1e8) / 1e8;
+    realExtent[2] = Math.round(realExtent[2] * 1e8) / 1e8;
```

Comment:
```suggestion
    realExtent[0] = Math.round(realExtent[0] * resolution) / resolution;
    realExtent[2] = Math.round(realExtent[2] * resolution) / resolution;
```
This would round the extent to pixels, which should be accurate enough.